components/eamxx/src/diagnostics/tests/extraaci_test.cpp

```diff
@@ -0,0 +1,184 @@
+  scorpio::read_var(iop_file, "lev", iop_pm.data());
+  scorpio::read_var(iop_file, "T", iop_tm.data(), time_idx);
+  scorpio::read_var(iop_file, "q", iop_qv.data(), time_idx);
+  scorpio::read_var(iop_file, "CLDLIQ", iop_qc.data(), time_idx);
+  scorpio::release_file(iop_file);
+  scorpio::finalize_subsystem();
+
+  // Construct random number generator stuff
+  using RPDF = std::uniform_real_distribution<Real>;
+  RPDF pdf(0, 0.05);
+  auto engine = scream::setup_random_test();
+
+  // Construct the Diagnostics
+  std::map<std::string, std::shared_ptr<AtmosphereDiagnostic>> diags;
+  auto &diag_factory = AtmosphereDiagnosticFactory::instance();
+  register_diagnostics();
+
+  constexpr int ntests = 1;
+  for(int itest = 0; itest < ntests; ++itest) {
+    // {
+    // Randomize
+    randomize(qc, engine, pdf);
+    randomize(qv, engine, pdf);
+    randomize(tm, engine, pdf);
+    randomize(pm, engine, pdf);
+    randomize(pd, engine, pdf);
+
+    // Create and set up the diagnostic
+    ekat::ParameterList params;
+    auto diag = diag_factory.create("ExtraAci", comm, params);
+    diag->set_grids(gm);
+    diag->set_required_field(qc);
+    diag->set_required_field(qv);
+    diag->set_required_field(tm);
+    diag->set_required_field(pm);
+    diag->set_required_field(pd);
+
+    diag->initialize(t0, RunType::Initial);
+
+    // Run diag
+    diag->init_timestep(t0);
+
+    randomize(qc, engine, pdf);
+    randomize(qv, engine, pdf);
+    randomize(tm, engine, pdf);
+    randomize(pm, engine, pdf);
+    randomize(pd, engine, pdf);
+
+    qc.get_header().get_tracking().update_time_stamp(t1);
+    qv.get_header().get_tracking().update_time_stamp(t1);
+    tm.get_header().get_tracking().update_time_stamp(t1);
+    pm.get_header().get_tracking().update_time_stamp(t1);
+    pd.get_header().get_tracking().update_time_stamp(t1);
+
+    auto tm_v = tm.get_view<Real **, Host>();
+    auto pm_v = pm.get_view<Real **, Host>();
+    auto qc_v = qc.get_view<Real **, Host>();
+    auto qv_v = qv.get_view<Real **, Host>();
+
+    for(int ilev = 1; ilev < nlevs; ++ilev) {
+      tm_v(0, ilev) = iop_tm[ilev];
+      pm_v(0, ilev) = iop_pm[ilev];
+      qc_v(0, ilev) = iop_qc[ilev];
+      qv_v(0, ilev) = iop_qv[ilev];
+    }
+
+    tm.sync_to_dev();
+    pm.sync_to_dev();
+    qc.sync_to_dev();
+    qv.sync_to_dev();
+
+    diag->compute_diagnostic();
+
+    // Check result
+    qc.sync_to_host();
+    qv.sync_to_host();
+    tm.sync_to_host();
+    pm.sync_to_host();
+
+    diag->get_diagnostic().sync_to_host();
+
+    const auto qc_h = qc.get_view<const Real **, Host>();
+    const auto qv_h = qv.get_view<const Real **, Host>();
+    const auto tm_h = tm.get_view<const Real **, Host>();
+    const auto pm_h = pm.get_view<const Real **, Host>();
+
+    const auto out_hf = diag->get_diagnostic();
+    auto out_hv       = out_hf.get_view<Real **, Host>();
+    Field out_tf      = diag->get_diagnostic().clone();
```

Comment:
I'm guessing this is leftover from previous iterations, since `out_t` is never really used...